gnomad_qc/v5/annotations/generate_frequency.py

```
@@ -0,0 +1,445 @@
+    )
+
+    # General arguments
+    parser.add_argument(
+        "--overwrite", help="Overwrite existing hail Tables.", action="store_true"
+    )
+
+    # Test/debug arguments
+    test_group = parser.add_argument_group("testing options")
+    test_group.add_argument(
+        "--data-test",
+        help="Filter to the first N partitions of full VDS for testing (N controlled by --test-partitions).",
+        action="store_true",
+    )
+    test_group.add_argument(
+        "--runtime-test",
+        help="Load test dataset and filter to test partitions.",
+        action="store_true",
+    )
+    test_group.add_argument(
+        "--test-partitions",
+        type=int,
+        default=2,
+        help="Number of partitions to use in test mode. Default is 2.",
+    )
+
+    # Processing step arguments
+    processing_group = parser.add_argument_group("processing steps")
+    processing_group.add_argument(
+        "--process-aou",
+        help="Process All of Us dataset for frequency calculations.",
+        action="store_true",
+    )
+    processing_group.add_argument(
+        "--use-all-sites-ans",
+        help="Use all sites ANs in frequency calculations to avoid a densify.",
+        action="store_true",
+    )
+
+    # Environment configuration
+    env_group = parser.add_argument_group("environment configuration")
+    env_group.add_argument(
+        "--environment",
+        help="Environment to run in.",
+        choices=["rwb", "batch", "dataproc"],
+        default="rwb",
+    )
+    env_group.add_argument(
+        "--tmp-dir-days",
+        type=int,
+        default=4,
+        help="Number of days for temp directory retention. Default is 4.",
+    )
+    env_group.add_argument(
+        "--gcp-billing-project",
+        type=str,
+        default="broad-mpg-gnomad",
+        help="Google Cloud billing project for reading requester pays buckets.",
+    )
+
+    # Batch-specific configuration
```

maybe there should be a check in main to to throw an error if any of these are set but environment is not set to batch?

---

Good idea, added!